identity/aziot-identityd/src/identity.rs

```diff
@@ -736,15 +751,33 @@ impl IdentityManager {
         &self,
         identity_pk: &str,
         identity_cert: &str,
-        subject: &str,
-    ) -> Result<(), Error> {
+        subject: Option<&str>,
+    ) -> Result<Option<Result<String, Error>>, Error> {
```

Comment:
Can this be flattened to just `Result<String, Error>`?

The `Ok(String)` value is the subject name, which must come from either the old cert or the config. Is there ever a scenario where the subject name won't be known by this function?

Reply:
The subject name doesn't need to exist in the manual X.509 provisioning case. Since this logic is shared between both DPS and IoT Hub, we need to late-evaluate this result.

Reply:
In X509 manual, this function is called with the subject parameter is set to `Some(&device_id)`. So even in that case, the subject name exists, right?

Reply:
The result isn't used in the manual x509 case. The `device_id` is only used to issue a new identity certificate (in case CS has EST method configured for `device-id` certificate).

Reply:
Which means it should be fine to flatten the return type to `Result<String, Error>`, right?
- In the manual X509 case, it's not used so doesn't matter
- In the DPS case, the parsing becomes simpler (currently you have nested matches to unwrap it).

Reply:
Flattened it to Result<Result<String, Error>> since we discussed the need for passing back a lazy-evaluated result.